src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -890,13 +923,14 @@ private void ExecuteBlockActions<TBlockAction>(
                 var codeBlockAction = blockAction as CodeBlockAnalyzerAction;
                 if (codeBlockAction != null)
                 {
+                    // This context is a struct, so it's fine to create a new one for each action.
```

Comment:
It looks like we're not lifting this out because we don't know which one we will need on each iteration? Is that right?

Reply:
correct.  but in a followup this will be lifted out as these iterations will be separated in a strongly typed fashion for the node vs iop cases.